spec/steps/follow_the_link_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "steps/follow_the_link"
+
+describe Spreewald::Steps::FollowTheLink do
+  it "raises helpful error message if no link is found" do
+    mail_without_links = Mail.new do
+      html_part do
+        body "<html><body>no link</body></html>"
+      end
+
+      text_part do
+        body "no link either"
+      end
+    end
+
+    step = ->() { Spreewald::Steps::FollowTheLink.new(mail_without_links, "first").run }
+    expect(step).to raise_error Spreewald::Steps::FollowTheLink::NoVisitableLinkFound
+  end
+
+  it "finds links in multipart html email" do
+    mail = Mail.new do
+      html_part do
+        body "<html><body><a href='https://www.example.com/abc'>a link</a></body></html>"
+      end
+
+      text_part do
+        body "no link here"
+      end
+    end
+
+    step = Spreewald::Steps::FollowTheLink.new(mail, "first") 
+    expect(step).to receive(:visit_path).with("/abc")
+    step.run
```

Comment:
I'm not quite sure if a `MailerSteps::LinkFinder` would be a better solution. So you could check the return value instead of the `visit` call.

Reply:
I think it would be. My plan for now is, whenever we touch a step that has no specs, we migrate it to a separate class, with as little changes as needed, spec it with a lot more edge cases. If we have migrated some, there will pop up some patterns we can extract (e.g. a Linkfinder).